server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -257,22 +262,15 @@ def from_tokenized(
             # request id -> idx in list mapping
             requests_idx_mapping[r.id] = i
 
-            tokenized_input = tokenized_input[-r.truncate :]
-            if (
-                tokenized_input[0] == tokenizer.bos_token_id
-                and tokenized_input[1] == tokenizer.bos_token_id
-            ):
-                tokenized_input = tokenized_input[1:]
-
             orig_input_length = len(tokenized_input)
 
-            if PREFIX_CACHING:
-                prefix_len = r.prefix_len
-                if prefix_len == orig_input_length:
-                    assert prefix_len > 0
-                    prefix_len -= 1
-            else:
-                prefix_len = 0
```

Comment:
Do we correctly default to 0 in the router?

Reply:
Yes, we do not however correctly account for leaving 1 token spare for the prefill to actually occur normally.